src/libsyntax/parse/parser.rs

```diff
@@ -1753,21 +1753,68 @@ impl<'a> Parser<'a> {
             (pat, self.parse_ty()?)
         } else {
             debug!("parse_arg_general ident_to_pat");
-            let ident = Ident::new(keywords::Invalid.name(), self.prev_span);
-            let ty = self.parse_ty()?;
-            let pat = P(Pat {
-                id: ast::DUMMY_NODE_ID,
-                node: PatKind::Ident(BindingMode::ByValue(Mutability::Immutable), ident, None),
-                span: ty.span,
-            });
-            (pat, ty)
+
+            // If we see `ident :`, then we know that the argument is not just of the
```

Comment:
`ident :` no longer needs to be treated specially because it falls under `self.is_named_argument()` above.

Reply:
In other words, we are already in cold code here and can make the snapshot unconditionally.